### DISCUSSION TITLE
react-bootstrap + react-hook-form + vest examples to provide ?

### DISCUSSION THREAD

**Discussion by jerome-diver** [react-hook-form/react-hook-form#6397] [General]
Hello and thank you for this wonderful library and for this space that we can ask questions.
I'm trying to make working together react-bootstrap (latest version) with react-hook-form and vest but i failed after 3 days of self investigations (and learning, because it's pretty new to me to use react-hook-form and vest). I also failed to find any examples on the web.
I'm running a SSR web site on my computer (still build it) and i want the validation to happen before to fetch to server (then i want it to operate before to submit, on change (or on blur) of any input field, and then not at submit time.
I miss-understand probably how react-hook-form handle the form data entry from... maybe from onHandleSubmit (that i'm thinking about but i'm not sure), and then how to make it handle data form fields without the need to submit anything ?

Please, can someone explain a bit around the process and provide an example with these technologies for a simple text input <Form.Control> inside an <Input.Group> maybe ?

Thank you very much to help. If i find something i will inform also there.

**Comment by bluebill1049** [comment]
can you provide a simple codesandbox of the issue that you are facing?

**Reply by jerome-diver** [reply]
It is not an issue report, but a question.
I just use now "useController" hook and "control" from "useForm" to go with react-bootstrap. 
But still not arrive to get dynamic validation on the go.
What i want is to update validation (not at submit time) from (maybe) onChange concerned input. 
So my problem is a lake of example use case to use:

- react-hook-form
- vest 
- react-bootstrap 
AND in **dynamic mode** (onChange input to act on validation process and to show error from validation with Form.Control.Feedback Component to print in view).

That is also why i'm asking for a use case example as simple as possible (sandbox or not) with the use of react-bootstrap Form.Control and Form.Control.Feedback Components in **dynamic mode**.

**Comment by jerome-diver** [comment]
I find something... i was far...
There is a "mode" at register time that do what i expect it to be: 'all'.  with this mode, the test is happening on blur and on change time.
And with the controller and formState, i can transmit form elements and errors back.
``` javascript
<Form onSubmit={handleSubmit(update(container))} 
            noValidate 
            validated={validated} 
            encType="multipart/form-data">
        <Form.Group>
            <Form.Label>{t('containers.update.title')}</Form.Label>
            <InputGroup hasValidation>
              <Controller name='title' control={control} render={
                ( {field: { onChange, onBlur, value, ref }, 
                   fieldState: { invalid }} ) => (
                <Form.Control type='text' onChange={onChange} ref={ref} isInvalid={invalid}
                              vaqlue={value} defaultValue={form.title[i18n.language]} /> )} />
              <Badge variant='info'>{t(container.type_name)}</Badge>
            <Form.Control.Feedback type="invalid">
              {t('containers.update.invalid_title')}
      {(formState.errors['title']) ? formState.errors.title.message : ''}
              </Form.Control.Feedback>
            </InputGroup>
            <Form.Text className="text-muted">{t('containers.helper.title')}</Form.Text>
        </Form.Group>
```
And i use the usual way to embed vest library to get validations rules separated as a test contract (the jest way) with "vest.only" on fieldname to get it fine.

Next question is to come about translation....
Thank you for your help @bluebill1049
